### PR TITLE
Migrate to opentelemetry 0.33 api metrics

### PR DESCRIPTION
**Pull request by richard-julien**

### CODE REVIEW

**Comment by marcbachmann** [comment]
Thanks for investing time in this.
Is this the only change that was needed to get it to work?

Could you change the commit message, so it will create a release automatically:
```
feat: Migrate to OpenTelemetry 0.33 api metrics

BREAKING CHANGE: Drop support for @opentelemetry/api-metrics < v0.33
```

Could you upgrade the example here: https://github.com/marcbachmann/opentelemetry-node-metrics/blob/master/example/package.json#L7-L8

If you don't have time, I can also directly commit to your branch.
I'll need to integrate it in our product anyways.

**Comment by richard-julien** [comment]
Hi @marcbachmann , 
Yes, not a lot of modification required, working well like that.
No problem for changing the commit message, I will do that tommorrow.

File: metrics/eventLoopLag.js [hunk 1]
```diff
@@ -32,43 +32,43 @@ module.exports = (meter, {prefix, labels, eventLoopMonitoringPrecision}) => {
 
   meter.createObservableGauge(prefix + NODEJS_EVENTLOOP_LAG_MIN, {
     description: 'The minimum recorded event loop delay.'
-  }, (observable) => {
+  }).addCallback(async (observable) => {
```
Comment (marcbachmann) [inline metrics/eventLoopLag.js]: Those callback functions don't need to use `async`.
It's optional according to the types: https://github.com/open-telemetry/opentelemetry-js/blob/d3adc22c4715d77e3a6913d0e14950fd68d470f9/experimental/packages/opentelemetry-api-metrics/src/types/Metric.ts#L104

 Please remove that here again.

**Comment by richard-julien** [comment]
Commit rewriten, you can take a new look.

**Review by marcbachmann — APPROVED** [review]
Looks good. Thanks.

**Comment by marcbachmann** [comment]
Regarding tests, I'm currently investigating that in another project.
It's a bit of a mess 🙃 , do you have some experience in testing some metric values?
```js
async function (metricName) => {
  const metrics = await prometheusExporter.collect()

  // Weirdly that can include errors, but they are never logged anywhere
  // Probably assert them in the tests
  for (const err of metrics.errors) console.error('Metric extract error', err)

  const metric = metrics.resourceMetrics.scopeMetrics?.[0]
    ?.find(m => m.descriptor.name === metricName)

  return metric.dataPoints?.[0]?.value
}
```

**Comment by richard-julien** [comment]
Sorry forget about your example and testing, i will take a look on that.

**Comment by marcbachmann** [comment]
There are no tests yet, I'm just wondering what's best practice there. I can probably setting up some later.
Manual testing is fine 👍

**Comment by richard-julien** [comment]
Commit rewritten with corret example. Honestly i have also no experience about testing it :)

File: example/package.json [hunk 2]
```diff
@@ -4,8 +4,8 @@
   "description": "",
   "main": "index.js",
   "dependencies": {
-    "@opentelemetry/exporter-prometheus": "^0.27.0",
-    "@opentelemetry/sdk-metrics-base": "^0.27.0",
+    "@opentelemetry/exporter-prometheus": "^0.33.0",
+    "@opentelemetry/sdk-metrics-base": "^0.31.0",
```
Comment (marcbachmann) [inline example/package.json]: That module got replaced by `@opentelemetry/sdk-metrics` 😉

**Pull request merged**